frontend/src/metabase-lib/lib/queries/StructuredQuery.js

```diff
@@ -1124,11 +1124,19 @@ export default class StructuredQuery extends AtomicQuery {
       const keyForFk = (src, dst) =>
         src && dst ? `${src.id},${dst.id}` : null;
       const explicitJoins = new Set(
-        joins.map(join => {
-          const p = join.parentDimension();
-          const j = join.joinDimension();
-          return keyForFk(p && p.field(), j && j.field());
-        }),
+        _.flatten(
+          joins.map(join => {
+            const dimensionPairs = join.getDimensions();
+            return dimensionPairs.map(pair => {
+              // eslint-disable-next-line no-unused-vars
+              const [parentDimension, joinDimension] = pair;
+              return keyForFk(
```

Comment:
I see in `keyForFk` that we are also doing falsy checks and setting this value to null if either value is falsy. (Assuming this is all pulled out into a separate function) I kind of think it might be more readable to inline the logic of `keyForFk` here so that we can rid ourselves of some of these checks.

```
parentDimension && joinDimension ? `${parentDimension.id},${joinDimension.id}` : null;
```

Reply:
`keyForFK` is used in two places (see [here](https://github.com/metabase/metabase/blob/1a062c36d9e5c8e2130acb8e770a6405a36203d1/frontend/src/metabase-lib/lib/queries/StructuredQuery.js#L1130) and [here](https://github.com/metabase/metabase/blob/1a062c36d9e5c8e2130acb8e770a6405a36203d1/frontend/src/metabase-lib/lib/queries/StructuredQuery.js#L1138)). Wdyt if we just extract it in a method?